pkg/controllers/nodeclaim/lifecycle/registration.go

```diff
@@ -88,12 +93,17 @@ func (r *Registration) syncNode(ctx context.Context, nodeClaim *v1beta1.NodeClai
 	// Sync all taints inside NodeClaim into the Node taints
 	node.Spec.Taints = scheduling.Taints(node.Spec.Taints).Merge(nodeClaim.Spec.Taints)
 	node.Spec.Taints = scheduling.Taints(node.Spec.Taints).Merge(nodeClaim.Spec.StartupTaints)
+	// Remove karpenter.sh/unregistered taint
+	taints := lo.Reject(node.Spec.Taints, func(t v1.Taint, _ int) bool {
```

Comment:
nit: any reason we can't just assign directly?
```suggestion
	node.Spec.Taints = lo.Reject(node.Spec.Taints, func(t v1.Taint, _ int) bool {
```